clippy_lints/src/redundant_pattern_matching.rs

```diff
@@ -103,14 +103,21 @@ fn find_sugg_for_if_let<'a, 'tcx>(
         arms[0].pat.span,
         &format!("redundant pattern matching, consider using `{}`", good_method),
         |diag| {
-            // in the case of WhileLetDesugar expr.span == op.span incorrectly.
-            // this is a workaround to restore true value of expr.span
-            let expr_span = expr.span.to(arms[1].span);
-            let span = expr_span.until(op.span.shrink_to_hi());
+            // while let ... = ... { ... }
+            // ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            let expr_span = arms[1].pat.span;
```

Comment:
Wait, what? I don't trust this at all. There is no test for `if let ... {} else if let {}`. I think this could break.

Reply:
> I think this could break.

I don't think so. I'll add this test.

Reply:
done